ctc_lib/ctc_lib.go

```diff
@@ -83,9 +84,9 @@ You can override it via ctc_lib.ConfigFile pkg variable`)
 }
 
 func initLogging() {
-	// Init File Logger
-	Log = logging.NewLogger(filepath.Join(viper.GetString("logDir"), toolName))
-	Log.SetLevel(flags.LogLevel.Level)
+	logging_dir, _ := ioutil.TempDir(viper.GetString("logDir"), toolName)
```

Comment:
I don't remember how viper.GetString handles defaults, what will be the default value here? I'd expect something like "", which then turns into the system default temp dir.

Reply:
The viper works is, 
1. Read first from config.
2. Read from flag default if flag not set or else Read the flag value.
```
(add_viper2)$ mkdir /tmp/logdir
###################### Log Dir set on Command line and Missing in the Viper Config
(add_viper2)$ bazel run demo/ctc:ctc_demo -- --verbosity=debug --logDir=/tmp/logdir
INFO: Running command line: bazel-bin/demo/ctc/linux_amd64_stripped/ctc_demo '--verbosity=debug' '--logDir=/tmp/logdir'
echo
See logs at  /tmp/logdir/echo684362576/debug.log.201803271700

###################### Log Dir not set on Command line and Missing in the Viper Config.
###################### Default Log Dir is not set to "". 
tejaldesai@tejaldesai~/go/src/github.com/GoogleCloudPlatform/runtimes-common: (add_viper2)$ bazel run demo/ctc:ctc_demo -- --verbosity=debug
INFO: Running command line: bazel-bin/demo/ctc/linux_amd64_stripped/ctc_demo '--verbosity=debug'
echo
See logs at  /tmp/echo657438736/debug.log.201803271700
tejaldesai@tejaldesai~/go/src/github.com/GoogleCloudPlatform/runtimes-common: (add_viper2)$ 
```
I had to change Default value for LogDir from /tmp" to ""  in 
1. https://github.com/tejal29/runtimes-common/blob/7799dda41c469b426d9ab8f157fdeeed209a3d6c/ctc_lib/container_tool_command_base.go#L73

2. DefaultConfig Defined at https://github.com/GoogleCloudPlatform/runtimes-common/pull/565/files#diff-58a4b875ce733b0e2844fc6e9553ebc3R20